tests/flows/erc20_source_erc20_destination_send_and_call.go

```diff
@@ -0,0 +1,228 @@
+	"github.com/ava-labs/teleporter-token-bridge/tests/utils"
+	"github.com/ava-labs/teleporter/tests/interfaces"
+	teleporterUtils "github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/crypto"
+	. "github.com/onsi/gomega"
+)
+
+/**
+ * Deploy an ERC20 token source on the primary network
+ * Deploys ERC20Destination to Subnet A
+ * Bridges C-Chain example ERC20 tokens to Subnet A
+ * Bridge tokens from Subnet A to C-Chain
+ */
+func ERC20SourceERC20DestinationSendAndCall(network interfaces.Network) {
+	cChainInfo := network.GetPrimaryNetworkInfo()
+	subnetAInfo, _ := teleporterUtils.GetTwoSubnets(network)
+	fundedAddress, fundedKey := network.GetFundedAccountInfo()
+
+	ctx := context.Background()
+
+	// Deploy an ExampleERC20 on the primary network as the source token to be bridged
+	sourceTokenAddress, sourceToken := teleporterUtils.DeployExampleERC20(
+		ctx,
+		fundedKey,
+		cChainInfo,
+	)
+
+	// Create an ERC20Source for bridging the source token
+	erc20SourceAddress, erc20Source := utils.DeployERC20Source(
+		ctx,
+		fundedKey,
+		cChainInfo,
+		fundedAddress,
+		sourceTokenAddress,
+	)
+
+	sourceMockERC20SACRAddress, _ := utils.DeployMockERC20SendAndCallReceiver(
+		ctx,
+		fundedKey,
+		cChainInfo,
+	)
+
+	destMockERC20SACRAddress, _ := utils.DeployMockERC20SendAndCallReceiver(
+		ctx,
+		fundedKey,
+		subnetAInfo,
+	)
+
+	// Token representation on subnet A will have same name, symbol, and decimals
+	tokenName, err := sourceToken.Name(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	tokenSymbol, err := sourceToken.Symbol(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	tokenDecimals, err := sourceToken.Decimals(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+
+	// Deploy an ERC20Destination to Subnet A
+	erc20DestinationAddress, erc20Destination := utils.DeployERC20Destination(
+		ctx,
+		fundedKey,
+		subnetAInfo,
+		fundedAddress,
+		cChainInfo.BlockchainID,
+		erc20SourceAddress,
+		tokenName,
+		tokenSymbol,
+		tokenDecimals,
+	)
+
+	// Generate new recipient to receive bridged tokens
+	recipientKey, err := crypto.GenerateKey()
+	Expect(err).Should(BeNil())
+	recipientAddress := crypto.PubkeyToAddress(recipientKey.PublicKey)
+
+	// Generate new recipient to receive bridged tokens
+	fallbackKey, err := crypto.GenerateKey()
+	Expect(err).Should(BeNil())
+	fallbackAddress := crypto.PubkeyToAddress(fallbackKey.PublicKey)
+
+	// Send tokens from C-Chain to recipient on subnet A
+	input := erc20source.SendAndCallInput{
+		DestinationBlockchainID:  subnetAInfo.BlockchainID,
+		DestinationBridgeAddress: erc20DestinationAddress,
+		RecipientContract:        destMockERC20SACRAddress,
+		RecipientPayload:         []byte{1},
+		RequiredGasLimit:         teleporterUtils.BigIntMul(big.NewInt(10), utils.DefaultERC20RequiredGasLimit),
+		RecipientGasLimit:        teleporterUtils.BigIntMul(big.NewInt(5), utils.DefaultERC20RequiredGasLimit),
```

Comment:
I was running out of gas, so I just chose large numbers. Is there a way to figure out what should be reasonable here?